### PR TITLE
[ONNX] Add symbolic for _convolution_mode

### PR DESCRIPTION
**Pull request by shubhambhokare1**
As per #68880
implement the operator _convolution_mode in the ONNX exporter. This will allow user to leverage the padding 'str' mode where it can be set to 'valid' or 'same'.

### CODE REVIEW

File: torch/onnx/symbolic_opset9.py [hunk 1]
```diff
@@ -2426,6 +2426,72 @@ def _convolution(
...
+    "v",
+    "v",
+    "v",
+    "is",
+    "s",
+    "is",
+    "i",
+)
+@_beartype.beartype
+def _convolution_mode(
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset9.py:2440]: Why not extend existing `_convolution` function?

Comment (shubhambhokare1) [reply torch/onnx/symbolic_opset9.py:2440]: Seems like aten::_convolution_mode is present in the IR graph as opposed to aten::_convolution in cases where padding is a string. Also the calling parameters for both these signatures are different.

File: torch/onnx/symbolic_opset9.py [hunk 2]
```diff
@@ -2460,75 +2526,117 @@ def convolution(
...
-        dilation,
-        False,
-        (),
-        groups,
-        None,
-        None,
-        None,
-        None,
-    )
+    str_padding = symbolic_helper._parse_arg(padding, "s")
```
Comment (AllenTiTaiWang) [inline torch/onnx/symbolic_opset9.py:2534]: Wondering if we need to parse_arg here, as we already have one on `convolution_mode` decorator?

File: test/onnx/test_pytorch_onnx_onnxruntime.py [hunk 3]
```diff
@@ -1167,6 +1167,27 @@ def forward(self, input1, input2, input3):
...
+    def test_conv_str_padding(self):
+        class TraceModel(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.conv1 = torch.nn.Conv1d(16, 33, 3, padding="valid")
+                self.conv2 = torch.nn.Conv2d(
+                    16, 33, (3, 5), stride=1, padding="valid", dilation=(3, 1)
+                )
+                self.conv3 = torch.nn.Conv3d(
+                    16, 33, (3, 5, 2), stride=1, padding="same"
```
Comment (AllenTiTaiWang) [inline test/onnx/test_pytorch_onnx_onnxruntime.py:1179]: nit: Consider using parallel tests on multiple parameters combinations (ex: conv3d with padding="valid") to avoid missing tests.

**Review by titaiwangms — APPROVED** [review]

**Review by BowenBao — APPROVED** [review]

**Comment by BowenBao** [comment]
@shubhambhokare1 rebase and merge?

**Comment by joasanna** [comment]
PLEASE MERGE THIS

**Comment by titaiwangms** [comment]
@shubhambhokare1 Can you merge this?

**Comment by kit1980** [comment]
@shubhambhokare1 @BowenBao should we merge this?

**Comment by Dawars** [comment]
@BowenBao The failing test should be fixed by https://github.com/pytorch/pytorch/issues/100400

**Comment by seamanj** [comment]
How can we use this feature? the latest version of torch on pypi is 2.0.0, which doesn't update. Also the latest Tag V2.0.1-rc2 doesn't update either. Could u update in the next release? thanks
